### PR TITLE
[v6r15] Data Logging System

### PR DESCRIPTION
**Pull request by coberger**
This is the data logging system for DIRAC. It allows to track all operations on files. It is transparent for users, it does not change the behavior of DIRAC. It is flexible for future additional.

Here is an example of the logging of a put and register operation and of a remove file operation :
$ python dirac/DIRAC/DataManagementSystem/scripts/dirac-dls-sequence.py /lhcb/sub/file2.data

Sequence 1, Caller dirac-dms-add-file.py , UserName coberger , HostName localhost.localdomain , Group devGroup
DataManager.putAndRegister, 2015-08-19 11:50:13, Successful, TargetSE StorageTest1
DataManager.registerFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1
FileCatalog.addFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1
StorageElement.putFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1

Sequence 2, Caller dirac-dms-remove-files.py , UserName coberger , HostName localhost.localdomain , Group devGroup
DataManager.removeFile, 2015-08-19 12:38:20, Successful
FileCatalog.removeFile, 2015-08-19 12:38:20, Successful
FileCatalog.removeReplica, 2015-08-19 12:38:20, Successful, TargetSE StorageTest1
StorageElement.removeFile, 2015-08-19 12:38:20, Successful, TargetSE StorageTest1

It is also possible to have all method calls about one LFN , here is the output:

$ python dirac/DIRAC/DataManagementSystem/scripts/dirac-dls-method-call.py /lhcb/sub/file2.data

Found 8 method calls
2015-08-19 11:50:13, DataManager.putAndRegister, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, DataManager.registerFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, StorageElement.putFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, FileCatalog.addFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 12:38:20, FileCatalog.removeFile, Successful, SequenceID 2
2015-08-19 12:38:20, DataManager.removeFile, Successful, SequenceID 2
2015-08-19 12:38:20, StorageElement.removeFile, TargetSE StorageTest1, Successful, SequenceID 2
2015-08-19 12:38:20, FileCatalog.removeReplica, TargetSE StorageTest1, Successful, SequenceID 2

It is possible to get method calls for one method name, it is also possible to precise a date before, a date after or an operation Status.

For sequences, it is possible to get all sequences for an LFN, a user name, a group name, an host name, etc.

### CODE REVIEW

File: DataManagementSystem/DB/DataLoggingDB.py [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+
+      :param lfn, a lfn name
+      :param name, a method name
+      :param before, a date, can be None
+      :param after, a date, can be None
+      :param status, a str in [ Failed, Successful, Unknown ], can be None
+
+      :return calls: a list of DLMethodCall
+    """
+    print 'toto %s' % errorCode
```
Comment (chaen) [inline DataManagementSystem/DB/DataLoggingDB.py]: To remove

Comment (coberger) [inline DataManagementSystem/DB/DataLoggingDB.py]: Remove, thank you

File: Core/Base/AgentModule.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def __init__( self, agentName, loadName, baseAgentName = False, properties = {}
     self.__configDefaults[ 'shifterProxy' ] = ''
     self.__configDefaults[ 'shifterProxyLocation' ] = os.path.join( self.__configDefaults[ 'WorkDirectory' ],
                                                                         '.shifterCred' )
-
+    os.environ['AGENTNAME'] = agentName
```
Comment (chaen) [inline Core/Base/AgentModule.py]: I have just seen that many agents define AGENT_NAME as a variable with a global scope. It is maybe a good idea to rationalize all this. Export it in the environment and set it as a global variable in the AgentModule does not seem unreasonable ?

**Comment by fstagni** [comment]
This PR can't be merged as it is.

**Comment by fstagni** [comment]
The releases.cfg file should not be here. 
At the same time there are commits not done by Corentin. This is not per se a problem, but seems there's some merging that was not properly done.

**Comment by chaen** [comment]
I replaced it with https://github.com/DIRACGrid/DIRAC/pull/2607
The fact that there are commits from others most probably means that he did a rebase and not a merge

**Pull request closed without merging**